mssqlcli/jsonrpc/contracts/tests/test_json_rpc_contracts.py

```diff
@@ -0,0 +1,249 @@
+#!/usr/bin/env python
+import io
+import os
+import time
+import unittest
+import mssqlcli.jsonrpc.contracts.connectionservice as connectionservice
+import mssqlcli.jsonrpc.contracts.queryexecutestringservice as queryservice
+import mssqlcli.jsonrpc.jsonrpcclient as json_rpc_client
+
+
+class JSONRPCContractsTests(unittest.TestCase):
+    """
+        Tests for validating SQLToolsservice API via JSONRPC.
+    """
+
+    def test_successful_connection_AdventureWorks2014(self):
+        """
+            Verify a successful connection response
+        """
+
+        with open(self.get_test_baseline(
+                  u'select_from_humanresources_department_adventureworks2014.txt'),
+                  u'r+b',
+                  buffering=0) as response_file:
+
+            request_stream = io.BytesIO()
+            rpc_client = json_rpc_client.JsonRpcClient(in_stream=request_stream,
+                                                       out_stream=response_file)
+            rpc_client.start()
+
+            owner_uri = u'connectionservicetest'
+            parameters = {u'OwnerUri': owner_uri,
+                          u'ServerName': u'bro-hb',
+                          u'DatabaseName': u'AdventureWorks2014',
+                          u'UserName': u'*',
+                          u'Password': u'*',
+                          u'AuthenticationType': u'Integrated',
+                          u'MultipleActiveResultSets': True}
+
+            request = connectionservice.ConnectionRequest(id=u'1',
+                                                          owner_uri=owner_uri,
+                                                          json_rpc_client=rpc_client,
+                                                          parameters=parameters)
+            self.verify_connection_service_response(request=request,
+                                                    expected_response_event=1,
+                                                    expected_complete_event=1)
+            rpc_client.shutdown()
+
+    def test_query_execute_response_AdventureWorks2014(self):

```

Comment:
>test_query_execute_response_AdventureWorks2014 [](start = 8, length = 46)

Can we add a test that does concurrent queries on two different owner uri?

Also, can we add any tests for non-happy path failure scenarios? For example, the refresh connection is killed while executing while the query connection continues.

Reply:
That's what this test does. It ensures that Request A never retrieves responses for Request B. This was the root cause of the display leak which is why I targeted this type of unit test. I can add a query connection failed test.

Reply:
Based on our discussion for restructuring the test story, I'll hold off on the failed connection test for now.